### PR TITLE
Inline popular functions

### PR DESCRIPTION
**Pull request by Komzpa**
Inline various popular functions similar to #290.

### CODE REVIEW

File: liblwgeom/lwinline.h [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PostGIS.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **********************************************************************
+ *
+ * Copyright 2018 Darafei Praliaskouski <me@komzpa.net>
```
Comment (dbaston) [inline liblwgeom/lwinline.h]: This is all copy-pasted from somewhere else; copyright should follow.

File: liblwgeom/lwkmeans.c [hunk 2]
```diff
@@ -157,11 +154,16 @@ kmeans_init(POINT2D** objs, int* clusters, uint32_t n, POINT2D** centers, POINT2
 		dst_p2 = distance2d_sqr_pt_pt(objs[i], objs[p2]);
 		if ((dst_p1 > max_dst) || (dst_p2 > max_dst))
 		{
-			max_dst = fmax(dst_p1, dst_p2);
 			if (dst_p1 > dst_p2)
+			{
```
Comment (dbaston) [inline liblwgeom/lwkmeans.c]: Are these changes related to the inlining?

**Comment by dbaston** [comment]
Seems reasonable in concept, but I'd be curious to know the performance difference in an actual query.

File: liblwgeom/lwkmeans.c (hunk 2 above)
Comment (Komzpa) [reply liblwgeom/lwkmeans.c]: yes, they remove a call to fmax/fmin.

**Comment by Komzpa** [comment]
On `-O3 -march=native -g` and gcc 7.3.0:

```
inlined:
-rwxrwxr-x 1 kom kom 3774184 вер 12 03:05 liblwgeom/.libs/liblwgeom-3.0.so.0.0.0*
stripped inlined:
-rwxrwxr-x 1 kom kom 825272 вер 12 03:40 liblwgeom/.libs/liblwgeom-3.0.so.0.0.0*

svn-trunk:
-rwxrwxr-x 1 kom kom 2719280 вер 12 03:37 liblwgeom/.libs/liblwgeom-3.0.so.0.0.0*
stripped svn-trunk:
-rwxrwxr-x 1 kom kom 641080 вер 12 03:38 liblwgeom/.libs/liblwgeom-3.0.so.0.0.0*
```
+180kb executable, +800kb debug symbols.

Speeds, inlined:

```
04:03:02 [postgis_reg] > select '#3971', count(*) 
[more] - > from (
[more] ( >          with
[more] ( >                  points as (
[more] ( >                  select ST_MakePoint(x, y) geom
[more] ( >                  from generate_series(1, 450) x, generate_series(1, 450) y
[more] ( >              )
[more] ( >          select
[more] ( >              ST_ClusterKMeans(geom, 81)
[more] ( >              over () as cid,
[more] ( >              geom
[more] ( >          from points
[more] ( >      ) z
[more] - > group by cid
[more] - > order by count(*)
[more] - > limit 1;
┌──────────┬───────┐
│ ?column? │ count │
├──────────┼───────┤
│ #3971    │  2077 │
└──────────┴───────┘
(1 row)

Time: 3468,420 ms (00:03,468)
```

2.5:
```
04:04:07 [kom] > select '#3971', count(*) 
[more] - > from (
[more] ( >          with
[more] ( >                  points as (
[more] ( >                  select ST_MakePoint(x, y) geom
[more] ( >                  from generate_series(1, 450) x, generate_series(1, 450) y
[more] ( >              )
[more] ( >          select
[more] ( >              ST_ClusterKMeans(geom, 81)
[more] ( >              over () as cid,
[more] ( >              geom
[more] ( >          from points
[more] ( >      ) z
[more] - > group by cid
[more] - > order by count(*)
[more] - > limit 1;
┌──────────┬───────┐
│ ?column? │ count │
├──────────┼───────┤
│ #3971    │  2077 │
└──────────┴───────┘
(1 row)

Time: 5942,388 ms (00:05,942)
```

**Comment by dbaston** [comment]
Impressive!

File: configure.ac [hunk 3]
```diff
@@ -58,6 +58,9 @@ dnl
 PICFLAGS="$lt_prog_compiler_pic"
 AC_SUBST([PICFLAGS])
 
+AC_LIBTOOL_COMPILER_OPTION([if $compiler supports -fno-math-errno], [_cv_nomatherrno], [-fno-math-errno], [], [CFLAGS="$CFLAGS -fno-math-errno"], [])
```
Comment (dbaston) [inline configure.ac]: Do these flags provide significant performance gain? I get some discomfort from venturing outside of IEEE floating point.

File: liblwgeom/lwinline.h (hunk 1 above)
Comment (Komzpa) [reply liblwgeom/lwinline.h]: fixed
